include/mxnet/tensor_blob.h

```diff
@@ -281,6 +291,34 @@ class TBlob {
     return this->get_with_shape<Device, 3, DType>(
         this->shape_.FlatTo3D(axis_begin, axis_end), stream);
   }
+
+ private:
+  static DLDataType DTypeTransform(int type_flag) {
+    static std::unordered_map<int, DLDataType>
+      MSHADOW_DTYPE_TO_DLPACK_DTYPE = {
+        {0, {2, 32, 1}},  // Float32
+        {1, {2, 64, 1}},  // Float64
+        {2, {2, 16, 1}},  // Float16
+        {3, {1,  8, 1}},  // UInt8
+        {4, {0, 32, 1}},  // Int32
+        {5, {0,  8, 1}}   // Int8
+      };
+    return MSHADOW_DTYPE_TO_DLPACK_DTYPE[type_flag];
+  }
+
+  inline void SetDLTensor() {
+    dltensor_.data = dptr_;
+    dltensor_.ctx = DLContext{-1, static_cast<DLDeviceType>(dev_mask_)};
```

Comment:
we can have a new argument defaults to -1
then modify all the places it was used in mxnet